### ISSUE TITLE
Language Code Search API Dash Search Problem

### ISSUE THREAD

**Issue by jag3773** [unfoldingWord-dev/door43.org#608]
In searching languages in our language page, I see that https://door43.org:9096/?q=pt-br does not return anything, where as https://door43.org:9096/?q=pt does have pt-br returned. Is there a reason for that? Is the dash a problem and somehow needs to be escaped?

The dash can be escaped but it is not necessary. It looks like it returns the same result either way, https://door43.org:9096/?q=pt%2Dbr. I'm guessing the problem is in the API Flask app.

Code is in https://github.com/unfoldingWord-dev/dw-lang-switcher

**Comment by richmahn** [comment]
Just looking at it real quick, I see that it has nothing to do with the hyphen. It doesn't search language code if the string is more than 2 characters...then it just searches the 'ang', 'lr' and 'ln' fields.

https://github.com/unfoldingWord-dev/dw-lang-switcher/blob/master/autocomplete.py#L112

**Comment by PhotoNomad0** [comment]
@richmahn @jag3773 - actually it will search language code if it is up to 3 characters https://github.com/unfoldingWord-dev/dw-lang-switcher/blob/master/autocomplete.py#L100, but still it is broken for longer language codes.  And in the case it is exactly 3 characters, it will do a second search https://github.com/unfoldingWord-dev/dw-lang-switcher/blob/master/autocomplete.py#L107

**Comment by PhotoNomad0** [comment]
Looking in the language json file I can see very long language codes (e.g. pna-x-punanhalanga).  Maybe we should just remove the length check in line 100?

**Comment by PhotoNomad0** [comment]
@richmahn @jag3773 - Or may be better to add the search terms `or term in x["lc"].lower() or term in [y.lower() for y in x["cc"]]` to line 112 and change line 100 to be `if len(term) <= 2:`

also the project is a good candidate for some unit tests. 😉

**Comment by richmahn** [comment]
Yeah, I would prefer to be able to at least type 8 characters. It is confusing if you search and it doesn't really change the results.
